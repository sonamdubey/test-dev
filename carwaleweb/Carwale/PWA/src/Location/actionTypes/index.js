export const OPEN_POPUP = 'OPEN_POPUP';
export const CLOSE_POPUP = 'CLOSE_POPUP';
export const SHOW_CROSS_ICON = 'SHOW_CROSS_ICON';
export const HIDE_CROSS_ICON = 'HIDE_CROSS_ICON';
