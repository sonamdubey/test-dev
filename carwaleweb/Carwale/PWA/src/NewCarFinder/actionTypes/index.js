export const NEXT_FILTER = 'NEXT_FILTER'
export const SET_CURRENT_SCREEN = 'SET_CURRENT_SCREEN'
export const SET_ALL_FILTER_SCREEN = 'SET_ALL_FILTER_SCREEN'

export const REQUEST_CAR_FEATURE = 'REQUEST_CAR_FEATURE'
export const RECEIVE_CAR_FEATURE = 'RECEIVE_CAR_FEATURE'
export const CAR_FEATURE_SELECTION = 'CAR_FEATURE_SELECTION'

// budget
export const UPDATE_BUDGET_SLIDER_VALUE = 'UPDATE_BUDGET_SLIDER_VALUE'
export const UPDATE_BUDGET_INPUT_VALUE = 'UPDATE_BUDGET_INPUT_VALUE'

// body type
export const REQUEST_BODY_TYPE = 'REQUEST_BODY_TYPE'
export const RECEIVE_BODY_TYPE = 'RECEIVE_BODY_TYPE'
export const BODY_TYPE_SELECTION = 'BODY_TYPE_SELECTION'
export const SET_BODY_STORED_FILTERS = 'SET_BODY_STORED_FILTERS'


// fuel type
export const REQUEST_FUEL_TYPE = 'REQUEST_FUEL_TYPE'
export const RECEIVE_FUEL_TYPE = 'RECEIVE_FUEL_TYPE'
export const FUEL_TYPE_SELECTION = 'FUEL_TYPE_SELECTION'
export const SET_FUEL_STORED_FILTERS = 'SET_FUEL_STORED_FILTERS'

// fuel type
export const REQUEST_MAKE = 'REQUEST_MAKE'
export const RECEIVE_MAKE = 'RECEIVE_MAKE'
export const MAKE_SELECTION = 'MAKE_SELECTION'
export const SET_MAKE_STORED_FILTERS = 'SET_MAKE_STORED_FILTERS'



// listing
// model list RECEIVE_MODEL_LIST
export const REQUEST_MODEL_LIST = 'REQUEST_MODEL_LIST'
export const RECEIVE_MODEL_LIST = 'RECEIVE_MODEL_LIST'
export const APPEND_MODEL_LIST = 'APPEND_MODEL_LIST'

// filter screen
export const OPEN_FILTER_SCREEN = 'OPEN_FILTER_SCREEN'
export const CLOSE_FILTER_SCREEN = 'CLOSE_FILTER_SCREEN'

// compare cars
export const ADD_MODEL_TO_COMPARE = 'ADD_MODEL_TO_COMPARE'
export const REMOVE_MODEL_FROM_COMPARE = 'REMOVE_MODEL_FROM_COMPARE'
// header
export const MAKE_HEADER_STICKY = 'MAKE_HEADER_STICKY'
export const MAKE_HEADER_NORMAL = 'MAKE_HEADER_NORMAL'
