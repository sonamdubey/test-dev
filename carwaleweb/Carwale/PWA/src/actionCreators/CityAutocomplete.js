import {
	SELECT_CITY,
	SELECT_AREA,
	SELECT_LATLONG,
	FETCHING_LOCATION,
	SELECT_AREA_CITIES,
	AUTODETECT_LOCATION
} from '../actionTypes'
import { initToast } from '../actionCreators/Toast'

const get = (url) => {
	let fetchUrl = url
	return fetch(fetchUrl, { headers: { sourceId: "43" } })
		.then(response => {
			if (!response.ok) {
				throw response
			}
			return response.json();
		})
}

export const setCity = ({ cityId, cityName, cityMaskingName, isConfirmBtnClicked = false }) => {
	return {
		type: SELECT_CITY,
		cityId,
		cityName,
		cityMaskingName,
		isConfirmBtnClicked
	}
}

export const setArea = ({ areaId, areaName }) => {
	return {
		type: SELECT_AREA,
		areaId,
		areaName,
	}
}

export const setLatLong = ({ latitude, longitude, autoDetect }) => {
	return {
		type: SELECT_LATLONG,
		latitude,
		longitude,
		autoDetect
	}
}
export const selectAreaCities = (areaCities) => {
	return {
		type: SELECT_AREA_CITIES,
		areaCities
	}
}
export const setAutoDetectLocation = (payload) => {
	return {
		type: AUTODETECT_LOCATION,
		payload
	}
}
const processAutoDetect = (position, dispatch) => {
	dispatch(setIsFetching(true))
	get('/api/locations/nearest/?latitude=' + position.coords.latitude + '&longitude=' + position.coords.longitude)
		.then(response => {
			if (response && response.length) {
				dispatch(setAutoDetectLocation({ ...response[0], ...position, autoDetect: true }));//need both response and position coord
			}
			dispatch(setIsFetching(false))
		})
		.catch(error => {
			errorprocessAutoDetect(dispatch);
		})
}
const errorprocessAutoDetect = (dispatch) => {
	dispatch(setIsFetching(false))
	dispatch(setLatLong({ autoDetect: false }))
	dispatch(initToast({
		message: 'Your location was not found!'
	}))
}
const setIsFetching = (isFetching) => {
	return {
		type: FETCHING_LOCATION,
		isFetching
	}
}
export const detectLocation = (pos) => (dispatch) => {
	if (pos) {
		processAutoDetect(pos, dispatch);
		return;
	}
	if (navigator.geolocation) {
		navigator.geolocation.getCurrentPosition(function (position) { processAutoDetect(position, dispatch) }, function (err) { errorprocessAutoDetect(dispatch) });
	} else {
		errorprocessAutoDetect(dispatch)
	}
}
