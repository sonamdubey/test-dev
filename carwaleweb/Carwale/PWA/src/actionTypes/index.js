// location: city
export const SELECT_CITY = "SELECT_CITY";
export const SELECT_AREA = "SELECT_AREA";
export const SET_LOCATION = "SET_LOCATION";
export const PREFILL_LOCATION = "PREFILL_LOCATION";

// toast
export const SET_TOAST = "SET_TOAST";
export const REMOVE_TOAST = "REMOVE_TOAST";

// header
export const SHOW_HEADER = "SHOW_HEADER";
export const HIDE_HEADER = "HIDE_HEADER";

// footer
export const SHOW_FOOTER = "SHOW_FOOTER";
export const HIDE_FOOTER = "HIDE_FOOTER";

// navigation drawer
export const OPEN_NAV_DRAWER = "OPEN_NAV_DRAWER";
export const CLOSE_NAV_DRAWER = "CLOSE_NAV_DRAWER";

// fullscreen
export const HIDE_HEADER_FOOTER = "HIDE_HEADER_FOOTER";
export const SHOW_HEADER_FOOTER = "SHOW_HEADER_FOOTER";

//page
export const SET_PAGE = "SET_PAGE";

// autoDetect location
export const AUTODETECT_LOCATION = "AUTODETECT_LOCATION"

export const SELECT_LATLONG = "SELECT_LATLONG";
export const FETCHING_LOCATION = "FETCHING_LOCATION"
export const SELECT_AREA_CITIES = "SELECT_AREA_CITIES"
