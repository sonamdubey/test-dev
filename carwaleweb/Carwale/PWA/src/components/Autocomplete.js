import React from "react";
import PropTypes from "prop-types";

import AutocompleteNoResult from "./AutocompleteNoResult";

const PropTypeReactComponent = PropTypes.oneOfType([
  PropTypes.func,
  PropTypes.string
]);

const propTypes = {
  // Input properties
  inputProps: PropTypes.object,
  // Input reference
  inputRef: PropTypes.func,
  // Input events
  inputEvents: PropTypes.object,
  // Suggestion list
  list: PropTypes.array,
  // Callback function fired on suggestion list item click
  onSuggestionClick: PropTypes.func,
  // Set focus to input field
  setFocus: PropTypes.bool,
  // Set clear button for input field
  setClearButton: PropTypes.bool,
  // Custom 'No result found' component
  noResultComponent: PropTypeReactComponent,
  // Display no result message
  noResultMessage: PropTypes.string
};

const defaultProps = {
  setClearButton: true,
  noResultComponent: AutocompleteNoResult
};

class Autocomplete extends React.Component {
  constructor(props) {
    super(props);

    this.blurTimer = null;

    this.state = {
      isListActive: false
    };
  }

  componentDidMount() {
    if (this.props.setFocus) {
      this.inputField.focus();
    }
  }

  componentWillUnmount() {
    clearTimeout(this.blurTimer);
  }

  componentWillReceiveProps(nextProps) {
    if (
      this.inputField === document.activeElement &&
      nextProps.inputProps.value &&
      nextProps.list
    ) {
      this.setState({
        isListActive: true
      });
    }
    if (nextProps.setFocus) {
      this.inputField.focus();
    }
  }

  handleInputFocus = () => {
    const { onFocus } = this.props.inputEvents;

    const { value } = this.props.inputProps;

    const { list } = this.props;

    if (value.length && list.length) {
      this.setState({
        isListActive: true
      });
    }

    if (onFocus) {
      onFocus();
    }
  };

  handleInputChange = event => {
    const { onChange } = this.props.inputEvents;

    const value = event.target.value;

    this.setState({
      isListActive: !!value.length && this.props.list.length
    });

    if (onChange) {
      onChange(event);
    }
  };

  handleInputKeyDown = event => {
    const { onKeyDown } = this.props.inputEvents;

    if (onKeyDown) {
      onKeyDown(event);
    }
  };

  handleInputBlur = () => {
    const { onBlur } = this.props.inputEvents;

    this.blurTimer = setTimeout(() => {
      this.setState({
        isListActive: false
      });
    }, 100);

    if (onBlur) {
      onBlur();
    }
  };

  handleInputClear = () => {
    const { onClear } = this.props.inputEvents;

    this.setState({
      isListActive: false
    });

    if (onClear) {
      onClear(this.inputField);
    }
  };

  handleSuggestionMouseDown = (item) => {
    const { onSuggestionClick } = this.props;

    /*
     * According to event order `onBlur` event gets fired before `onClick`,
     * `onClick` callback function was not getting fired on input blur event.
     * Trigger click event with onMouseDown event.
     * https: //stackoverflow.com/questions/17769005/onclick-and-onblur-ordering-issue
     */
    if (onSuggestionClick) {
      onSuggestionClick(item)
    }
  }

	onClick = () => {
		const {
			onClick,
		} = this.props.inputEvents

		if (onClick) {
			onClick()
		}
	}

setReference = ref => {
    const { inputRef } = this.props;
    this.inputField = inputRef ? inputRef(ref) : ref;
  };

  renderSuggestionList = () => {
    const { list } = this.props;

    const styleList = this.state.isListActive ? "show" : "hide";

    if (list) {
      return (
        <ul className={"autocomplete-list " + styleList}>
          {this.suggestionListItem()}
        </ul>
      );
    }
  };

  suggestionListItem = () => {
    const { list, isFetching } = this.props;

    if (list.length) {
      let listElement = list.map((item, index) => {
        return (
          <li key={index} className="autocomplete-list-item">
            <div
              className="autocomplete-list-item__label"
              onMouseDown={this.handleSuggestionMouseDown.bind(this, item)}
            >
              {this.highlighter(item.result)}
            </div>
          </li>
        );
      });

      return listElement;
    } else if (!isFetching) {
      const { noResultComponent: NoResult, noResultMessage } = this.props;

      return <NoResult message={noResultMessage} />;
    }
  };

  highlighter = result => {
    const { value } = this.props.inputProps;

    let escapedPhrase = value.replace(
      /[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g,
      "\\$&"
    );
    let tokens = result.split(new RegExp("(" + escapedPhrase + ")", "ig"));
    const regx = new RegExp(escapedPhrase, "i");
    let final = tokens.reduce((acc, token, index) => {
      return acc.concat(
        regx.test(token) ? (
          <strong key={index}>{token}</strong>
        ) : (
          <span key={index}>{token}</span>
        )
      );
    }, []);

    return final;
  };

  render() {
    const { inputProps, setClearButton } = this.props;

    const styleClearTarget = inputProps.value.length ? "show" : "hide";

    return (
      <div className="autocomplete-box">
        <div className="autocomplete-field">
          <input
            type="text"
            ref={this.setReference}
            id={inputProps.id ? inputProps.id : ""}
            placeholder={inputProps.placeholder ? inputProps.placeholder : ""}
            className={inputProps.className ? inputProps.className : ""}
            value={inputProps.value}
            onChange={this.handleInputChange}
            onFocus={this.handleInputFocus}
            onBlur={this.handleInputBlur}
            onKeyDown={this.handleInputKeyDown}
            onClick={this.onClick}
            autoComplete="off"
          />
          {setClearButton && (
            <span
              className={"clear-field-target " + styleClearTarget}
              onClick={this.handleInputClear}
            >
              Clear
            </span>
          )}
        </div>
        {this.renderSuggestionList()}
      </div>
    );
  }
}

Autocomplete.defaultProps = defaultProps;
Autocomplete.propTypes = propTypes;

export default Autocomplete;
