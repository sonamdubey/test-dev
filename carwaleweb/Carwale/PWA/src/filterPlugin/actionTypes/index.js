export const OPEN_FILTER_SCREEN = 'OPEN_FILTER_SCREEN'
export const CLOSE_FILTER_SCREEN = 'CLOSE_FILTER_SCREEN'
export const SET_FILTERS = 'SET_FILTERS'
