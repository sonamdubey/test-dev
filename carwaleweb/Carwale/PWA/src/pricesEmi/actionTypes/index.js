// budget
export const UPDATE_MODEL_DATA= 'UPDATE_MODEL_DATA'
export const UPDATE_DOWNPAYMENT_SLIDER_VALUE = 'UPDATE_DOWNPAYMENT_SLIDER_VALUE'
export const UPDATE_DOWNPAYMENT_INPUT_VALUE = 'UPDATE_DOWNPAYMENT_INPUT_VALUE'
export const UPDATE_TENURE_SLIDER_VALUE = 'UPDATE_TENURE_SLIDER_VALUE'
export const UPDATE_TENURE_INPUT_VALUE = 'UPDATE_TENURE_INPUT_VALUE'
export const UPDATE_INTEREST_SLIDER_VALUE = 'UPDATE_INTEREST_SLIDER_VALUE'
export const UPDATE_INTEREST_INPUT_VALUE = 'UPDATE_INTEREST_INPUT_VALUE'
export const UPDATE_PIE_VALUE = 'UPDATE_PIE_VALUE'
export const SHOW_EMI_POPUP = 'SHOW_EMI_POPUP'
export const HIDE_EMI_POPUP = 'HIDE_EMI_POPUP'

// set model id for emi
export const SET_EMI_MODEL = 'SET_EMI_MODEL'
export const UPDATE_EMI_MODEL = 'UPDATE_EMI_MODEL'
export const SET_DOWNPAYMENT_SNAPPOINT = 'SET_DOWNPAYMENT_SNAPPOINT'
export const SET_IS_EMI_VALID = 'SET_IS_EMI_VALID'
