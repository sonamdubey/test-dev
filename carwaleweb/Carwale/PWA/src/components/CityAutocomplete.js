import React from 'react'
import PropTypes from 'prop-types'
import { throttle } from 'throttle-debounce'

import Autocomplete from '../components/Autocomplete'

import autocomplete from '../utils/Autocomplete'

import {
  makeCancelable
} from '../utils/CancelablePromise'

import {
  CITY_AUTOCOMPLETE_URL
} from '../constants'

const propTypes = {
  // Input field id
  id: PropTypes.string,
  // Input field class
  className: PropTypes.string,
  // Input field placeholder
  placeholder: PropTypes.string,
  // Object with city details
  cityLocation: PropTypes.object,
  // Callback function gets fired on input change
  onChange: PropTypes.func,
  // Set clear button for input field
  setClearButton: PropTypes.bool,
  // Callback function gets fired on input clear
  onClear: PropTypes.func,
  // Callback function gets fired on clear button click
  clearSelection: PropTypes.func,
  // Callback function gets fired on city selection
  setCity: PropTypes.func,
  // Confirm button
  showConfirmBtn: PropTypes.bool,
  // Is Location Popup active
  isActive:PropTypes.bool
}

const defaultProps = {
  id: 'getCity',
  className: 'city-field',
  placeholder: 'Type your city',
  setClearButton: true,
  isActive: true
}

/**
 * This Class is a container component for the {@link Autocomplete} component
 * to be used in location screen.
 * @class CityAutocomplete
 * @extends {React.Component}
 */

 //The component is exported as unconnected component for custom handling
 export class CityAutocomplete extends React.Component {
  constructor(props) {
    super(props)
    this.apiRequest = null
    this.state = {
      input: props.cityLocation.cityName.trim() === 'Select City' ? '' : props.cityLocation.cityName,
      cityList: [],
      isFetching: false,
    }
    this.fetchCities = throttle(500, this.fetchCities);
  }

  componentWillReceiveProps = (nextProps) =>{
    if (this.props.cityLocation !== nextProps.cityLocation || !nextProps.isActive) {
      this.setState({
        input: nextProps.cityLocation.cityName.trim() === 'Select City' ? '' : nextProps.cityLocation.cityName,
      })
    }
  }

  componentWillUnmount() {
    this.apiRequest && this.apiRequest.cancel()
  }

  onChange = event => {
    const {
      onChange,
    } = this.props;

    const value = event.target.value

    this.setState({
      input: value,
      isFetching: true,
    })

    this.fetchCities(value)

    if(onChange) {
      onChange(event);
    }
  }

  /**
   * Method to make request for cities
   * and wrap the promise as cancelable
   * and set it in state.
   * @memberof CityAutocomplete
   * @param {string} input User input value
   */
  fetchCities = input => {
    if (input) {
      let apiRequest = makeCancelable(autocomplete({
        url: CITY_AUTOCOMPLETE_URL,
        source: 43,
        resultCount: 5
      }, input))
      apiRequest
        .promise
        .then((json) => this.afterFetch(json))
        .catch(({isCanceled, ...error}) => console.log('isCanceled', isCanceled));
      this.apiRequest = apiRequest
    }
  }

  afterFetch = (list) => {
    if(this.state.input && list){
      this.setState({
        cityList: list,
        isFetching: false,
      })
    }
  }

  onClear = (inputField) => {
    const {
      clearSelection,
      onClear,
    } = this.props;

    this.setState({
      input: '',
      cityList: [],
    }, inputField.focus())

    if (clearSelection) {
      clearSelection()
    }

    if (onClear) {
      onClear()
    }
  }

  handleSuggestionClick = (item) => {
    const {
      setCity,
    } = this.props

    const {
      cityId,
      cityName,
      cityMaskingName,
    } = item.payload

    let citySelection = {
      cityId,
      cityName,
      cityMaskingName,
      isConfirmBtnClicked: true
    }

    this.setState({
      input: cityName,
    })
    setCity(citySelection)
  }

  render() {
    const {
      cityList,
      input,
      isFetching,
    } = this.state;

    const {
      inputRef,
      setClearButton,
      onKeyDown,
      showConfirmBtn,
      onFocus,
      onBlur,
    } = this.props

    const inputProps = {
      ...defaultProps,
      value: input,
    }

    const inputEvents = {
      onChange: this.onChange,
      onClear: this.onClear,
      onKeyDown: onKeyDown,
      onClick: this.props.onClick,
      onFocus: onFocus,
      onBlur: onBlur,
    }

    return (
      <Autocomplete
        inputRef={inputRef}
        inputProps={inputProps}
        inputEvents={inputEvents}
        list={cityList}
        isFetching={isFetching}
        onSuggestionClick={this.handleSuggestionClick}
        setFocus={!showConfirmBtn}
        setClearButton={setClearButton}
        noResultMessage="No city found!"
      />
    )
  }
}

CityAutocomplete.propTypes = propTypes
CityAutocomplete.defaultProps = defaultProps

export default CityAutocomplete
