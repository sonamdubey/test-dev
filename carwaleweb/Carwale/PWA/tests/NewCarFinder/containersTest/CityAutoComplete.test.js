import React from 'react'
import { Redirect } from 'react-router-dom'

import CityAutocomplete from '../../../src/components/CityAutocomplete'
import Autocomplete from '../../../src/components/Autocomplete'

import { SELECT_CITY } from '../../../src/actionTypes'


describe("Tests for CityAutocomplete container component",()=>{

    let state
    let props
    let store
    let enzymeWrapper

    const Wrapper = () => {
        if(!enzymeWrapper){
            store = mockStore(state)
            enzymeWrapper = shallowWithStore(<CityAutocomplete {...props}/>, store)
                            .dive() // dive() because it is a connected component
        }
        return enzymeWrapper
    }

    beforeEach(()=>{
        props = undefined
        state = undefined
        enzymeWrapper = undefined
        store = undefined
    })

    it("should render w/o confirm when no city cookie present", () => {
        state = {
            location:{
                cityId:-1,
                cityName:'Select City'
            }
        }

        const wrapper = Wrapper()

        expect(wrapper.find(Autocomplete)).toHaveLength(1)
        expect(wrapper.find('.btn-confirm-city')).toHaveLength(0)
        expect(wrapper).toMatchSnapshot()

    })

    it("should render confirm when city cookie present", () => {
        state = {
            location:{
                cityId:1,
                cityName:'Mumbai, Maharashtra'
            }
        }

        const wrapper = Wrapper()

        expect(wrapper.find(Autocomplete)).toHaveLength(1)
        expect(wrapper.find('.btn-confirm-city')).toHaveLength(1)
        expect(wrapper).toMatchSnapshot()

    })

    it("should redirect to success url on pressing confirm", () => {
        state = {
            location:{
                cityId:1,
                cityName:'Mumbai, Maharashtra'
            }
        }

        props = { successUrl : '/find-car/filters'}

        const wrapper = Wrapper()

        expect(wrapper.find('.btn-confirm-city')).toHaveLength(1)

        expect(wrapper.find(Redirect)).toHaveLength(0)

        wrapper.find('.btn-confirm-city').simulate('click')

        expect(wrapper.find(Redirect)).toHaveLength(1)

        const expectedActions = store.getActions()
        expect(expectedActions).toHaveLength(1)

        expect(expectedActions).toContainEqual({
            type: SELECT_CITY,
            cityId: state.location.cityId,
            cityName: state.location.cityName,
            isConfirmBtnClicked: true
        })

        expect(wrapper).toMatchSnapshot()
    })

    it("should redirect to success url on list item selection", () => {
        state = {
            location:{
                cityId:-1,
                cityName:''
            }
        }

        props = { successUrl : '/find-car/filters'}

        const wrapper = Wrapper()

        const item = {
            result:"Mumbai, Maharashtra",
            payload:{
               cityId:1,
               cityName:"Mumbai"
           }
        }

        expect(wrapper.find(Redirect)).toHaveLength(0)

        wrapper.instance().onListItemSelection(item)

        // To update the shallow render tree @reference: https://github.com/airbnb/enzyme/issues/450
        wrapper.update()

        expect(wrapper.find(Redirect)).toHaveLength(1)
        const expectedActions = store.getActions()
        expect(expectedActions).toHaveLength(1)

        expect(expectedActions).toContainEqual({
            type: SELECT_CITY,
            cityId: item.payload.cityId,
            cityName: item.payload.cityName,
            isConfirmBtnClicked: true
        })

        expect(wrapper).toMatchSnapshot()
    })
})
