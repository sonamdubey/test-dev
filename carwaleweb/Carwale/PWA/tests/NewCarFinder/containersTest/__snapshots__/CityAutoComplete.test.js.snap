// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for CityAutocomplete container component should redirect to success url on list item selection 1`] = `
<Redirect
  push={true}
  to={
    Object {
      "pathname": "/find-car/filters",
    }
  }
/>
`;

exports[`Tests for CityAutocomplete container component should redirect to success url on pressing confirm 1`] = `
<Redirect
  push={true}
  to={
    Object {
      "pathname": "/find-car/filters",
    }
  }
/>
`;

exports[`Tests for CityAutocomplete container component should render confirm when city cookie present 1`] = `
<div
  className="location__content"
>
  <div
    className="location-content__head"
  >
    <p
      className="location-selection__title"
    >
      Confirm
       your City
    </p>
    <p
      className="location-selection__subtitle"
    >
      This is necessary to personalise results for you
    </p>
  </div>
  <div
    className="location-content__body"
  >
    <Autocomplete
      inputEvents={
        Object {
          "onBlur": [Function],
          "onChange": [Function],
          "onClear": [Function],
          "onFocus": [Function],
        }
      }
      inputProps={
        Object {
          "className": "city-field",
          "id": "getCity",
          "placeholder": "Type your city",
          "successUrl": "/",
          "value": "Mumbai, Maharashtra",
        }
      }
      list={Array []}
      renderSuggestionList={[Function]}
      setFocus={false}
    />
    <button
      className="btn-secondary btn-confirm-city"
      onClick={[Function]}
    >
      Confirm
    </button>
  </div>
</div>
`;

exports[`Tests for CityAutocomplete container component should render w/o confirm when no city cookie present 1`] = `
<div
  className="location__content"
>
  <div
    className="location-content__head"
  >
    <p
      className="location-selection__title"
    >
      Select
       your City
    </p>
    <p
      className="location-selection__subtitle"
    >
      This is necessary to personalise results for you
    </p>
  </div>
  <div
    className="location-content__body"
  >
    <Autocomplete
      inputEvents={
        Object {
          "onBlur": [Function],
          "onChange": [Function],
          "onClear": [Function],
          "onFocus": [Function],
        }
      }
      inputProps={
        Object {
          "className": "city-field",
          "id": "getCity",
          "placeholder": "Type your city",
          "successUrl": "/",
          "value": "",
        }
      }
      list={Array []}
      renderSuggestionList={[Function]}
      setFocus={true}
    />
  </div>
</div>
`;
